### PR TITLE
7853 deprecation warnings 1

### PR DESCRIPTION
**Pull request by apricot12**
#### What? Why?

Closes parts of #7853

This PR just fixes some of the deprecatin warnings found througout the application.

Note: Most of the `file.exists?` deprecation warnings are unrelated to our application and are from the gems we use.

#### What should we test?
- Run the related spec files and see if the deprecation warnings still exist.
- Green build

#### Release notes
- Fixes some deprecation warnings.

Changelog Category:  Technical changes

### CODE REVIEW

File: app/models/spree/payment.rb [hunk 1]
```diff
@@ -181,15 +181,25 @@ def revoke_adjustment_eligibility
       )
     end
 
+    # rubocop:disable Metrics/AbcSize, Metrics/LineLength
     def validate_source
       if source && !skip_source_validation && !source.valid?
-        source.errors.each do |field, error|
-          field_name = I18n.t("activerecord.attributes.#{source.class.to_s.underscore}.#{field}")
-          errors.add(Spree.t(source.class.to_s.demodulize.underscore), "#{field_name} #{error}")
+        if Gem::Requirement.new(">= 6.1").satisfied_by?(Rails.gem_version)
```
Comment (mkllnk) [inline app/models/spree/payment.rb:187]: Why do we need this? Our current Rails version is 6.1.4 and we are not planning on going backwards.

File: app/models/spree/payment.rb [hunk 2]
```diff
@@ -181,15 +181,25 @@ def revoke_adjustment_eligibility
       )
     end
 
+    # rubocop:disable Metrics/AbcSize, Metrics/LineLength
```
Comment (mkllnk) [inline app/models/spree/payment.rb:184]: It's better to refactor than introducing these lines. Or, if it's really out of scope, then add this file to the rubocop_todo list instead.

**Review by mkllnk — COMMENTED** [review]
Nice! Some really good changes in here.

File: app/models/spree/payment.rb (hunk 2 above)
Comment (apricot12) [reply app/models/spree/payment.rb:184]: I'll try and refactore that then 👍

**Review by mkllnk — APPROVED** [review]
Much easier. :+1:

**Comment by sigmundpetersen** [comment]
> Closes https://github.com/openfoodfoundation/openfoodnetwork/issues/7853

Will this close the issue or will there be follow-up PRs?

**Comment by apricot12** [comment]
> > Closes https://github.com/openfoodfoundation/openfoodnetwork/issues/7853
> 
> 
> 
> Will this close the issue or will there be follow-up PRs?

There needs to be a follow up PR. I've noticed a few more places with depreciations...

**Comment by sigmundpetersen** [comment]
> > > Closes #7853
> > 
> > 
> > Will this close the issue or will there be follow-up PRs?
> 
> There needs to be a follow up PR. I've noticed a few more places with deprecations...

OK I just edited the description so it doesn't close the issue automatically when merged 🙂

**Review by jibees — APPROVED** [review]
Looks good to me! ;)

**Pull request merged**